### ISSUE TITLE
Some thoughts after running and inspecting TiDB

### ISSUE THREAD

**Issue by unw9527** [xlab-uiuc/acto#134]
Just write down some thoughts after running and inspecting half of TiDB results so that everyone can see.

1. A problem we might want to solve is, when setting the number of clusters to 2, we cannot get 2x speed. When running tidb, one thread still have more than 300 test cases to process while the other thread has already finished (and the early finished thread became idle afterwards, leaving only one thread working for two more days). 
 **One possible solution** is to randomize test cases to make the complexity/failing rate of cases similar on the two threads. I will try to implement this before starting the next run of tidb. But I am not sure how much this solution can help. Welcome any further suggestions on this.
3. It seems that I cannot run TiDB using 4 clusters. When running with 4 clusters, 2 clusters are never up and eventually throw a runtime error. Right now, running with 2 clusters almost costs a week to finish all test cases. I am not sure whether this is a known limitation of our multi-cluster design. I could share the log if anyone wants to get some insight.
4. Another issue is about the implicit requirement: for creating pods, only `replicas` are explicitly required in the tidb crd. However, `storage` is also required, which is not in the crd. This causes more than 100 alarms in Acto, as Acto changes only `replicas`, which results in no change in the system state. Tidb also throws no errors in the log indicating that `storage` is required. I have done some experiments running Acto with a seed that explicitly adds both `storage` and `replicas`. And it seems that the change in `replicas` can be reflected in the system state under this setting. I remember @tylergu said before that implicit requirement is a grey zone and probably worth discussing with prof @tianyin before any conclusion is made.

PS. It seems that one tidb’s developer has replied to the [PR](https://github.com/pingcap/tidb-operator/pull/4608) I submitted previously. However, I am not sure whether his suggestion is correct, since his suggestion will only raise an error if `MaxFailoverCount>0`. But even when `MaxFailoverCount=0`, we probably want to raise an error as long as `pdDeletedFailureReplicas>0`. What do you think? @tylergu
